ste/JobPartPlan.go

```diff
@@ -399,6 +404,26 @@ func (jppt *JobPartPlanTransfer) ErrorCode() int32 {
 	return atomic.LoadInt32(&jppt.atomicErrorCode)
 }
 
+// ErrorMessage returns the transfer's error message.
+func (jppt *JobPartPlanTransfer) ErrorMessage() string {
+	return string(jppt.errorMessage[:jppt.errorMessageLength])
+}
+
+// SetErrorMessage sets the error message if transfer failed.
+// overWrite flags if set to true overWrites the errorMessage.
+// If overWrite flag is set to false, then errorMessage won't be overwritten.
+func (jppt *JobPartPlanTransfer) SetErrorMessage(errorMessage string, overwrite bool) {
+	if !overwrite {
+		if jppt.errorMessageLength == 0 {
+			jppt.errorMessageLength = int32(len(errorMessage))
+			copy(jppt.errorMessage[:], []byte(errorMessage))
+		}
+	} else {
+		jppt.errorMessageLength = int32(len(errorMessage))
+		copy(jppt.errorMessage[:], []byte(errorMessage))
+	}
+}
+
```

Comment:
do it like this. This way we are safe from two goroutines messing the size and log in inconsistent ways which can lead to crashes etc

`func (jppt *JobPartPlanTransfer) SetErrorMessage(errorMessage string, overwrite bool) {
    savedErrorMessageLength := jppt.errorMessageLength

    // Overwrite, if this is the first error or caller wants this new errorMessage to overwrite the existing one.
    if (savedErrorMessageLength == 0) || overwrite {
        if sync.CompareAndSwapInt32(&jppt.errorMessageLength, savedErrorMessageLength, int32(len(errorMessage))) {
            copy(jppt.errorMessage[:], []byte(errorMessage))
        }
    }
}`

Reply:
sorry, the indentation is messed up

Reply:
i used atomic ops for errorMessageLength, does it always handle inconsistent reads/writes ? since we are not doing atomic ops on errorMessage ?